tests/test_utils.py

```diff
@@ -163,3 +170,425 @@ def test_parse_log_components(ns_3_compiled_debug, config):
+            'Severity_class': 'DEBUG',
+            'Message': 'Saving channel number configuration for initialization'
+        },
+        {
+            'Time': 0.506390387,
+            'Context': '1',
+            'Extended_context': 'mac=00:00:00:00:00:01',
+            'Component': 'FrameExchangeManager',
+            'Function': 'StartTransmission',
+            'Arguments': '',
+            'Severity_class': 'DEBUG',
+            'Message': 'MPDU payload size=36, to=00:00:00:00:00:02, seq=16'
+        },
+        {
+            'Time': 0.000000000,
+            'Context': '-1',
+            'Extended_context': None,
+            'Component': 'WifiPhy',
+            'Function': 'SetChannelNumber',
+            'Arguments': '',
+            'Severity_class': 'DEBUG',
+            'Message': '()[]./-='
+        }
+    ]
+
+    assert len(parse_list) == 6
+    assert all([actual == expected for actual, expected in zip(parse_list, expected_list)])
+
+    os.remove(data_dir)
+
+
+def test_filters():
+    db_dir = os.path.join(os.curdir, 'tests', str(time.strftime("%Y-%m-%d::%H-%M-%S")) + ':test_logs.json')
+    data_dir = os.path.join(os.curdir, 'tests', str(time.strftime("%Y-%m-%d::%H-%M-%S")) + ':test_logs')
+
+    db = TinyDB(db_dir,
+                storage=CachingMiddleware(JSONStorage))
+
+    log_ls = ['+0.000000000s -1 [mac=00:00:00:00:00:00] FrameExchangeManager:SetWifiMac(0x5576595683e0, 0x557659603820)\n',
+              '+0.045510017s 1 [mac=00:00:00:00:00:01] FrameExchangeManager:RxStartIndication(0x5576595683e0, "PSDU reception started for ", +76us, " (txVector: ", txpwrlvl: 17 preamble: LONG channel width: 20 GI: 800 NTx: 97 Ness: 0 MPDU aggregation: 0 STBC: 0 FEC coding: BCC mode: OfdmRate6Mbps Nss: 1, ")")\n',
+              '+0.506390387s 1 [mac=00:00:00:00:00:01] FrameExchangeManager:StartTransmission(): [DEBUG] MPDU payload size=36, to=00:00:00:00:00:02, seq=16\n',
+              '+2.999074264s 0 [mac=00:00:00:00:00:02] FrameExchangeManager:NotifyReceivedNormalAck(0x56239764d8c0, DATA, payloadSize=1456, to=00:00:00:00:00:01, seqN=2323, duration/ID=+44000ns, lifetime=+195539us, packet=0x56239786dd00)\n',
+              '+0.000000000s -1 WifiPhy:WifiPhy(0x557659571bd0)\n',
+              '+0.000000000s -1 WifiPhy:SetChannelNumber(): [DEBUG] Saving channel number configuration for initialization\n',
+              '+0.000000000s -1 WifiPhy:SetChannelNumber(): [INFO] ()[]./-=\n',
+              '+2.999825333s 1 WifiPhy:GetTxPowerForTransmission(): [DEBUG ] txPowerDbm=17 with txPowerDbmPerMhz=3.9897 over 20 MHz\n',
+              '+2.999825333s 1 WifiPhy:GetTxPowerForTransmission(): [INFO ] txPowerDbm=17 after applying m_powerDensityLimit=100\n'
+              ]
+
+    with open(data_dir, "w") as f:
+        f.writelines(log_ls)
+
+    parse_list = parse_logs(data_dir)
+    insert_logs(parse_list, db)
+
+    filter_list = filter_logs(db, context=['1'])
+    assert len(filter_list) == 4
+    assert filter_list == [
```

Comment:
assuming expected_list is a global variable
```suggestion
    filter_list = filter_logs(db, context=['1'])
    expected_filter_list = [x for x in expected_list if x['Context']==1] # make sure that this works
    assert len(filter_list) == len(expected_filter_list)
    assert filter_list == expected_filter_list # i'm not sure that you can compare lists of dicts, but i think you get the point of this suggestion
```

Reply:
Corrected in c8e8db3.